extension-style/src/main/java/com/mapbox/maps/extension/style/utils/TypeUtils.kt

```diff
@@ -44,6 +44,7 @@ internal object TypeUtils {
         Value(value)
       }
       is Double -> {
+        require(!value.isNaN()) { "Value can not be Double.NaN" }
```

Comment:
Should we do the same for `Float` below? and their array counterparts?

Reply:
Actually, the same will happen if INFINITY is passed.
See Moshi tests: https://github.com/square/moshi/blob/master/moshi/src/test/java/com/squareup/moshi/MoshiTest.java#L296

Reply:
Fixed, please take a look again.